cryptol-remote-api/python/cryptol/quoting.py

```diff
@@ -29,9 +29,8 @@ def to_cryptol_str(val : Union[CryptolValue, str, CryptolJSON]) -> str:
     elif isinstance(val, OpaqueValue):
         return str(val)
     elif isinstance(val, str):
-        # Sanitizing strings for Cryptol - see <string> of src/Cryptol/Parser/Lexer.x
-        val_san = val.replace('\\', '\\\\').replace('"', '\\"').replace('\n', '\\n')
-        return f'"{val_san}"'
+        chars = list(val.encode('latin-1'))
+        return f'({to_cryptol_str(chars)} : [{len(chars)}][8])'
```

Comment:
I think this look good now.

While we are fixing things here, could you also fix the instance for BV a few lines above?  I think it would do the wrong thing for 0-bit vectors, in that it would render them as `0x0` which would then be interpreted as a 4-bit vector by Cryptol.    You'd just need to change the conditional to `val.size() > 0 && val.size() % 4 == 0`

Reply:
Fixed both - thanks @yav!

Also looks like the Cryptol pretty-printer makes the same mistake that I did for `[0]`:
```
Cryptol> 0 : [0]
0x0
Cryptol> 0x0 : [0]

[error] at <interactive>:5:1--5:4:
  Type mismatch:
    Expected type: 0
    Inferred type: 4
    When checking user annotation
```
Finally I noticed the following – is `0x` the preferred way to represent the inhabitant of `[0]`?
```
Cryptol> :t 0x
0x0 : [0]
```

Reply:
Ah okay, good to know about the way printing on the REPL works. And I definitely agree with your suggestion of falling back on the else case for 0-bit values (I already implemented it in 93b3d79f3cbc0d11e2806d5400a2d903402f473d).

I think the PR is waiting on you to give the OK now – so if there's nothing else, let me know when you approve and I can merge it!